SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java

```diff
@@ -368,16 +382,27 @@ CheckTaskState getTaskState(SingularityTask task, Optional<SingularityRequestWit
     return Optional.absent();
   }
 
-  private boolean isOverdue(SingularityTask task) {
-    final long taskDuration = System.currentTimeMillis() - task.getTaskId().getStartedAt();
+  private boolean isHealthcheckOverdue(SingularityTask task) {
+    final long healthcheckDuration = taskManager.getLastHealthcheck(task.getTaskId()).get().getDurationMillis().or(0L);
```

Comment:
which time are you referring to? `task.getTaskid().getStartedAt()` looked like launch time. i couldn't find anything that shows the time is updated once it starts running. I also don't see any other times except for system time